apps/aechannel/src/aesc_close_solo_tx.erl

```diff
@@ -216,28 +216,45 @@ serialization_template(?CHANNEL_CLOSE_SOLO_TX_VSN) ->
                            ok | {error, term()}.
 check_payload(ChannelId, FromPubKey, Payload, PoI, Trees) ->
-    case aesc_utils:deserialize_payload(Payload) of
-        {ok, last_onchain} ->
-            case aesc_utils:get_active_channel(ChannelId, Trees) of
-                {ok, Channel} ->
+    case aesc_utils:get_channel(ChannelId, Trees) of
+        {error, _} = E -> E;
+        {ok, Channel} ->
+            case aesc_utils:deserialize_payload(Payload) of
+                {ok, last_onchain} ->
                     Checks =
-                          [fun() -> check_channel_hash(Channel, PoI) end,
-                           fun() ->
-                               aesc_utils:check_peers_and_amounts_in_poi(Channel,
-                                                                         PoI)
-                           end],
+                        [fun() -> check_root_hash_in_channel(Channel, PoI) end,
+                         fun() -> check_peers_and_amounts_in_poi(Channel, PoI) end
+                        ],
                     aeu_validation:run(Checks);
-                {error, _} = Err -> Err
-            end;
-        {ok, SignedState, PayloadTx} ->
-            Checks =
-                [fun() -> is_peer(FromPubKey, SignedState, Trees) end,
-                 fun() -> aetx_sign:verify(SignedState, Trees) end,
-                 fun() -> check_channel(ChannelId, PayloadTx, PoI, Trees) end,
-                 fun() -> check_root_hash(PayloadTx, PoI) end],
-            aeu_validation:run(Checks);
-        {error, _Reason} = Error ->
-            Error
+                {ok, SignedState, PayloadTx} ->
+                    Checks =
+                        [fun() -> check_channel_id_in_payload(Channel, PayloadTx) end,
+                         fun() -> check_round_in_payload(Channel, PayloadTx) end,
+                         fun() -> check_root_hash_in_payload(PayloadTx, PoI) end,
+                         fun() -> is_peer(FromPubKey, SignedState, Trees) end,
+                         fun() -> aetx_sign:verify(SignedState, Trees) end,
+                         fun() -> check_peers_and_amounts_in_poi(Channel, PoI) end
+                        ],
+                    aeu_validation:run(Checks);
+                {error, _Reason} = Error ->
+                    Error
+            end
+    end.
+
+check_peers_and_amounts_in_poi(Channel, PoI) ->
+    InitiatorPubKey   = aesc_channels:initiator(Channel),
+    ResponderPubKey   = aesc_channels:responder(Channel),
+    ChannelAmount     = aesc_channels:total_amount(Channel),
+    case aesc_utils:accounts_in_poi([InitiatorPubKey, ResponderPubKey], PoI) of
+        {error, _} = Err -> Err;
+        {ok, [PoIInitiatorAcc, PoIResponderAcc]} ->
+            PoIInitiatorAmt = aec_accounts:balance(PoIInitiatorAcc),
+            PoIResponderAmt = aec_accounts:balance(PoIResponderAcc),
+            PoIAmount       = PoIInitiatorAmt + PoIResponderAmt,
+            case ChannelAmount =:= PoIAmount of
```

Comment:
Would it not be acceptable here with `ChannelAmount >= PoIAmount`?

Reply:
Yes. This is a restriction that we have discussed to lift, but I aimed at not changing the semantics for now.